lib/suse/connect/zypper.rb

```diff
@@ -176,7 +176,7 @@ def write_service_credentials(service_name)
         def remove_service_credentials(service_name)
           service_credentials_file = File.join(SUSE::Connect::Credentials::DEFAULT_CREDENTIALS_DIR, service_name)
 
-          if File.exist?(service_credentials_file)
+          if File.file?(service_credentials_file) && File.exist?(service_credentials_file)
```

Comment:
`File.file?` should be enough. It can only validate that `service_credentail_file` is a regular file if it can actually stat the file.